src/database/mod.rs

```diff
@@ -158,6 +168,73 @@ pub trait Database: BatchOperations {
     ///
     /// It should insert and return `0` if not present in the database
     fn increment_last_index(&mut self, keychain: KeychainKind) -> Result<u32, Error>;
+
+    /// Delete UTXOs provided as params, if `spent_utxos` is empty it deletes all spent
+    /// UTXOs in the database.
+    fn del_spent_utxos(
+        &mut self,
+        spent_outpoints: Option<Vec<OutPoint>>,
```

Comment:
probably better to name it something like `to_delete`? All utxos in this context are "spent".

Reply:
@rajarshimaitra don't you think it is more descriptive to keep `del_spent_utxos`?

Reply:
I think he means using `to_delete` instead of `spent_outpoints`

Reply:
Yes that's what I meant..